be/src/storage/delete_handler.cpp

```diff
@@ -267,6 +275,7 @@ OLAPStatus DeleteHandler::init(const TabletSchema& schema, const DelPredicateArr
         temp.filter_version = it->version();
 
         temp.del_cond = new (std::nothrow) Conditions();
+        ScopedCleanup del_cond_delete_guard([&]() { delete temp.del_cond; });
```

Comment:
It's more common to use `std::unique_ptr` as a memory release guard.

Reply:
I see, `_del_conds.push_back` may throw an exception, and if so, a memory leak may happen if we are using a std::unique_ptr as the memory release guard. Thanks for pointing it out @laotan332.
It's so difficult and unmaintainable to write exception-safe code, maybe we should rethink do we really need exception-safe.